src/ring_theory/graded_algebra/homogeneous_localization.lean

```diff
@@ -68,22 +68,22 @@ variables {ι R A: Type*}
 variables [add_comm_monoid ι] [decidable_eq ι]
 variables [comm_ring R] [comm_ring A] [algebra R A]
 variables (𝒜 : ι → submodule R A) [graded_algebra 𝒜]
-variables (x : ideal A) [ideal.is_prime x]
+variables (x : submonoid A)
 
-local notation `at ` x := localization.at_prime x
+local notation `at ` x := localization x
 
 namespace homogeneous_localization
 
 section
 /--
-Let `x` be a prime ideal, then `num_denom_same_deg 𝒜 x` is a structure with a numerator and a
+Let `x` be a submonoid of `A`, then `num_denom_same_deg 𝒜 x` is a structure with a numerator and a
 denominator with same grading such that the denominator is not contained in `x`.
 -/
 @[nolint has_nonempty_instance]
 structure num_denom_same_deg :=
 (deg : ι)
 (num denom : 𝒜 deg)
-(denom_not_mem : (denom : A) ∉ x)
+(denom_wd : (denom : A) ∈ x)
```

Comment:
I'm not sure what `wd` means here?

Reply:
well-defined

Reply:
I see. I don't think it is intuitive though. I think something like `denom_mem` would be better?

Reply:
The problem is that `denom_mem` already says `denom \in x` so another `denom_mem'` might become confusing, how about `denom_mem_submonoid`?

Reply:
I would be inclined to rename the existing `denom_mem` (and probably `num_mem`) to `denom_mem_deg` or `denom_deg` instead, since `denom_mem_submonoid` seems a bit long for me.

Reply:
done in [8ddd7ce](https://github.com/leanprover-community/mathlib/pull/16653/commits/8ddd7cef37690b2b995a458f781f11611bd7feb2)

Reply:
The docstring in the start of the file also needs to be changed. Also, one of `homogeneous_localization.num` should be `homogeneous_localization.denom` in that docstring.

Reply:
sorry for the late reply. Docstrings are fixed in [bcefdf8](https://github.com/leanprover-community/mathlib/pull/16653/commits/bcefdf8b1fdbdca575867d482a07615e23057ea8)